src/ResourceManagement/ServerManagement/ServerManagement.Tests/ServerManagementTestBase.cs

```diff
@@ -0,0 +1,271 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Security.Principal;
+using System.ServiceProcess;
+using System.Threading.Tasks;
+using Microsoft.Azure.Management.ServerManagement;
+using Microsoft.Azure.Test.HttpRecorder;
+using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+using Xunit.Abstractions;
+
+
+namespace ServerManagement.Tests
+{
+    public class ServerManagementTestBase : TestBase
+    {
+        protected static string ResourceGroup = "sdktest";
+        private static bool _once;
+
+        private readonly ITestOutputHelper _output;
+
+        protected static bool Recording
+        {
+            get { return HttpMockServer.Mode == HttpRecorderMode.Record; }
+        }
+
+        protected static bool ReuseExistingGateway
+        {
+            get
+            {
+                return Recording &&
+                       "true".Equals(Environment.GetEnvironmentVariable("SMT_REUSE_EXISTING_GATEWAY"),
```

Comment:
Please add some comments around usage of this env var, in case the next person recording the test is someone else